pkg/icingadb/history/retention.go

```diff
@@ -177,12 +178,21 @@ func (r *Retention) Start(ctx context.Context) error {
 			fmt.Sprintf("Starting history retention for category %s", stmt.Category),
 			zap.Uint64("count", r.count),
 			zap.Duration("interval", r.interval),
-			zap.Uint16("retention-days", days),
+			zap.Float32("retention-days", days),
 		)
 
 		stmt := stmt
 		periodic.Start(ctx, r.interval, func(tick periodic.Tick) {
-			olderThan := tick.Time.AddDate(0, 0, -int(days))
+			olderThan := tick.Time
+			wholeDays, dayFraction := math.Modf(float64(days))
+
+			if wholeDays > 0 {
```

Comment:
Can this be negative? Above in line 171/172 it is already checked if `days < 1`.

Btw, you may want to change the abort constraint to `days < 0` as otherwise you cannot use half days, right?

Reply:
> Can this be negative?

I guess you answered your own question:

> Above in line 171/172 it is already checked if `days < 1`.

So no, it can't be.

(If I understand you correctly.)